shell/platform/darwin/ios/framework/Source/FlutterTextInputPlugin.mm

```diff
@@ -2488,7 +2500,22 @@ - (void)setSelectionRects:(NSArray*)encodedRects {
                                                          ? NSWritingDirectionLeftToRight
                                                          : NSWritingDirectionRightToLeft]];
   }
+
+  BOOL shouldNotifyTextChange = NO;
+  if (@available(iOS 17, *)) {
+    // Force UIKit to query the selectionRects again on iOS 17+
+    // This is to fix a bug on iOS 17+ where UIKit queries the outdated selectionRects after
```

Comment:
Similar to above, I think this bug exists on all versions of iOS, right?

Reply:
Not in this case - in iOS 16, we do not use system highlight, so no such bug. 

I also tried to notify the change on iOS 16 and nothing seems to break, but just to be safe here.